optimade.rst

```diff
@@ -442,42 +442,62 @@ For example, the following query can be sent to API implementations `exmpl1` and
 
 :filter:`filter=_exmpl1_band_gap<2.0 OR _exmpl2_band_gap<2.5`
 
-Per property metadata fields
-----------------------------
-
-Implementations are allowed to specify an OPTIONAL field containing per entry metadata for a property.
-The name of the metadata field consists of the name of the property for which it contains the metadata, suffixed with "_meta".
-For example, when the field is :property:`cartesian_site_positions` the metadata field would be :field:`cartesian_site_positions_meta`.
+Metadata properties
+-------------------
 
-This metadata field consists of a dictionary which MAY contain database specific fields.
-If an implementation supports the metadata field, it SHOULD return the metadata field whenever the property to which the metadata field belongs is returned.
+A metadata property represents property-specific metadata for each entry.
+How these are communicated in the response depend on the response format.
+For the JSON response format, the resource object metadata field is used, see `JSON Response Schema: Common Fields`_.
 
-The metadata fields and their subfields should be described in the property definitions as described in the section `property definitions`_ just as regular fields.
-If a subfield is present in multiple metadata fields, these subfields should have a separate entry under each of these metadata fields.
-The subfields SHOULD have the same value for the :field:`$id` field if the :field:`$id` field is present and the subfields are otherwise identical.
-In addition to the fields described in the section `property definitions`_, the property definition of a metadata field SHOULD have the key:
+The metadata property is a dictionary of format specified by the field :field:`x-optimade-metadata-definition` in the Property Definition of the field, see `Property Definitions`_.
+Database providers are allowed to define their own metadata properties in :field:`x-optimade-metadata-definition`, but they MUST use the database-specific prefix even for metadata for database-specific fields.
+For example, the metadata property definition of the field :field:`_exmpl_example_field` MUST NOT define a metadata field named, e.g., :field:`accuracy`; the field rather needs to be named, e.g., :field:`_exmpl_accuracy`.
+The reason for this limitation is to avoid name collisions with standard metadata fields defined by the OPTIMADE standard that apply also to database-specific data fields.
 
-- **x-optimade-metadata-for**: This key contains a string with the name of the property for which this metadata property contains the metadata.
-  It MUST be located at the top level of the property definition.
+If an implementation supports the metadata field, it SHOULD return the metadata field whenever the property to which the metadata field belongs is returned.
```

Comment:
How does this work in the case of one entry not having the associated metadata? e.g., adding an entry below to which there is no confidence interval, or it is invalid, e.g., a database that serves a theoretical structure with exact `elements_ratios: [1]` alongside experimental structures that have confidence intervals? Should the database be forced to assign a value (I guess confidence interval of 1.0 in this case), or are they allowed to omit the metadata field? Further down we say `meta` here is a `MAY` per resource but does that contradict the "SHOULD" here?

Reply:
The "SHOULD" here is meant to refer to including, e.g., the field `meta -> element_ratios` in which you would then put things like confidence intervals. The property definition for that meta field should specify whether the confidence interval is required or optional to include in that meta field. Given what the metadata property definition says, an empty `meta -> element_ratios` should be fine.

I don't have strong feelings how we should put the MAY and SHOULDs here, but the way it is presently is meant to convey that implementations MAY support metadata fields (but MAY thus chose not to). However, IF they support them, they SHOULD always include them in the `meta` part of the entry response if the corresponding data field is included.